lib/phoenix_live_view/html_parser.ex

```diff
@@ -0,0 +1,453 @@
+defmodule Phoenix.LiveView.HTMLParser do
+  @moduledoc false
+
+  alias Phoenix.LiveView.Tokenizer
+
+  defguard is_tag_open(tag_type)
+           when tag_type in [:slot, :remote_component, :local_component, :tag]
+
+  # Reference for all inline elements so that we can tell the formatter to not
+  # force a line break. This list has been taken from here:
+  #
+  # https://developer.mozilla.org/en-US/docs/Web/HTML/Inline_elements#list_of_inline_elements
+  @inline_tags ~w(a abbr acronym audio b bdi bdo big br button canvas cite
+  code data datalist del dfn em embed i iframe img input ins kbd label map
+  mark meter noscript object output picture progress q ruby s samp select slot
+  small span strong sub sup svg template textarea time u tt var video wbr)
+
+  @inline_components ~w(.link)
+
+  @inline_elements @inline_tags ++ @inline_components
+
+  @doc false
```

Comment:
Would love to see the comments below consolidated into a short function doc, as these are super useful in IDEs